### PR TITLE
Allow disabling of invalid watchers

### LINKED ISSUE #90: Throwing from disable() with invalid watcher
I've been running into timing issues causing `Loop::disable()` to throw due to a watcher being invalidated during the invocation of the watcher callback. Essentially, if the watcher callback calls `Loop::disable()` at the end of the callback after side-effects from the watcher callback have invoked `Loop::cancel()`, an exception is being thrown. While it is possible to use an additional flag to code around this problem, it seems unnecessary.

Some example code: https://github.com/amphp/socket/blob/amp_v2/lib/Server.php#L37-L51

Putting the [call to `Loop::cancel()` in a defer watcher](https://github.com/amphp/socket/blob/amp_v2/lib/Server.php#L76-L78) _seems_ to solve the issue here. I tried the same in the `Socket` class within the same lib [here](https://github.com/amphp/socket/blob/amp_v2/lib/Socket.php#L200-L203). However, this does not appear to have solved the problem, as tests still occasionally result in [throws due to invalid watchers](https://travis-ci.org/amphp/concurrent/jobs/153426676#L418).

I propose that `InvalidWatcherException` should not be thrown from `disable()` with cancelled watchers, as disabling a cancelled watcher does not change the expectation that further events will not be received on the watcher. `enable()` should continue throwing when a cancelled watcher is given.

### PR DESCRIPTION
**Pull request by bwoebi**
Fixes #90.

Following no dissent in the related issue, this should be merged in a few days if nobody else objects.

### CODE REVIEW

File: src/Loop/Driver.php [hunk 1]
```diff
@@ -134,13 +134,12 @@
     abstract public function enable($watcherId);
 
     /**
-     * Disable a watcher. Disabling a watcher MUST NOT invalidate the watcher.
+     * Disable a watcher. Disabling a watcher MUST NOT invalidate the watcher. Calling this function MUST never fail,
```
Comment (kelunik) [inline src/Loop/Driver.php]: `MUST never` → `MUST NOT`

File: src/Loop/Driver.php [hunk 2]
```diff
@@ -134,13 +134,12 @@
...
+     * even when passed an invalid watcher.
```
Comment (kelunik) [inline src/Loop/Driver.php]: I guess this should be `if`, not `when`.

File: src/Loop/Driver.php (hunk 1 above)
Comment (bwoebi) [reply src/Loop/Driver.php]: I've copied from cancel() docs.

Comment (kelunik) [reply src/Loop/Driver.php]: Should probably fixed there as well. Otherwise you might miss the `never` since only `MUST` is bold.

**Pull request merged**